dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -101,39 +104,58 @@ def manual_pass
     redirect_to :manual_pass_form
   end
 
+  # get /admin/permissions
   def permissions_form
+    search_term = params[:search_term]
+    permission = params[:permission]
+    if search_term.present?
+      if search_term =~ /^\d+$/
+        @user = User.find(search_term)
+      else
+        users = User.where(hashed_email: User.hash_email(search_term))
+        @user = users.first
+        if users.count > 1
+          flash[:notice] = "More than one User matches email address.  "\
+                         "Showing first result.  Matching User IDs - #{users.map(&:id).join ','}"
+        end
+      end
+      unless @user || search_term.blank?
+        flash[:notice] = "User Not Found"
+      end
+    elsif permission.present?
+      @users_with_permission = User.joins(:permissions).where(user_permissions: {permission: permission}).order(:email)
```

Comment:
I added fields like sign_in_count and recent_sign_in_at to help an administrator figure out which user they want to grant/revoke permissions in the rare scenarios where there are multiple Users with the same email address.

I chose to use `select` instead of `pluck` because `select` returns a model (with only the specified attributes set) whereas `pluck` returns an array of unlabelled attribute values, that I'd have to convert to a hash in order to easily use them in the view.